__tests__/app.test.js

```diff
@@ -116,26 +116,88 @@ describe("GET /api/", () => {
+            const { status, body: { message } } = await request(app).get("/api/articles/apple/comments");
+            expect(status).toBe(400);
+            expect(message).toBe("apple is an invalid article_id (number)");
+        });
 
     });
 
+});
+describe("POST /api/", () => {
+    describe("/articles/:article_id/comments", () => {
+        test("201: status code and responds with body of comment inserted ", async () => {
+            const myComment = { username: "butter_bridge", body: "I didn't read this." };
+            const expectedResponse = {
+                "comment_id": 19,
+                "body": "I didn't read this.",
+                "article_id": 9,
+                "author": "butter_bridge",
+                "votes": 0,
+                "created_at": expect.any(String)
+            };
+            
+            const { status, body: {comment} } = await request(app).post("/api/articles/9/comments").send(myComment);
+
+            expect(status).toBe(201)
+            expect(comment).toEqual(expectedResponse)
+
+
+
+        });
+        test("404: status code when username is not found",async () => {
+            const myComment = { username: "anonymous", body: "I don't like this." }; 
+            const { status, body: {message} } = await request(app).post("/api/articles/9/comments").send(myComment);
+
+            expect(status).toBe(404)
+            expect(message).toBe("Couldn't find username anonymous!")
+        })
+        test("404: status code when article_id is not found",async () => {
+            const myComment = { username: "butter_bridge", body: "I don't like this." }; 
+            const { status, body: {message} } = await request(app).post("/api/articles/9999/comments").send(myComment);
+
+            expect(status).toBe(404)
+            expect(message).toBe("Couldn't find article_id 9999!")
+        })
+        test("404: status code when given wrong path",async () => {
+            const { status, body: {message} } = await request(app).post("/api/articels/9/comments")
+
+            expect(status).toBe(404)
+            expect(message).toBe("Path not found")
+        })
+        test("400: status code when body of request is missing username",async () => {
+            const myComment = { body: "I don't like this." }; 
+            const { status, body: {message} } = await request(app).post("/api/articles/9999/comments").send(myComment);
+
+            expect(status).toBe(400)
+            expect(message).toBe("Invalid request! Missing information!")
+        })
+        test("400: status code when body of request is missing body",async () => {
+            const myComment = { username: "butter_bridge"}; 
+            const { status, body: {message} } = await request(app).post("/api/articles/9999/comments").send(myComment);
+
+            expect(status).toBe(400)
+            expect(message).toBe("Invalid request! Missing information!")
+        })
```

Comment:
Really thorough testing here 😄 Nice to see you have accounted for a request that is missing required fields 👍